### PR TITLE
feat(components): New sagemaker training job parameters

### PR DESCRIPTION
**Pull request by ananth102**
**Description of your changes:**

Autogenerated/linter Changes:
```
components/aws/sagemaker/TrainingJob/component.yaml 
components/aws/sagemaker/TrainingJob/src/TrainingJob_component.py
components/aws/sagemaker/TrainingJob/src/TrainingJob_request.yaml.tpl
components/aws/sagemaker/TrainingJob/src/TrainingJob_spec.py
components/aws/sagemaker/commonv2/__init__.py
components/aws/sagemaker/commonv2/sagemaker_component.py
```

Manual change:
`components/aws/sagemaker/tests/integration_tests/scripts/run_integration_tests`

Testing method:
Created an ack image from updatetrain3 and deployed it.
InstanceGroups/Warmpool/normal job work with the resulting outputs showing up correctly.

Do not merge until new ack is released into helm.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: components/aws/sagemaker/TrainingJob/component.yaml [hunk 1]
```diff
@@ -179,11 +190,16 @@ outputs:
...
+  - {
+      name: warm_pool_status,
+      type: JsonObject,
+      description: "The status of the warm pool associated with the training job.",
+    }
   ###########################GENERATED SECTION ABOVE############################
 
 implementation:
   container:
-    image: public.ecr.aws/kubeflow-on-aws/aws-sagemaker-kfp-components:2.0.0
```
Comment (ryansteakley) [inline components/aws/sagemaker/TrainingJob/component.yaml:186]: make sure to change this before merging

**Review by ryansteakley — APPROVED** [review]

File: components/aws/sagemaker/TrainingJob/component.yaml (hunk 1 above)
Comment (ananth102) [reply components/aws/sagemaker/TrainingJob/component.yaml:186]: Yea

**Review by surajkota — COMMENTED** [review]
looks good, try to revert the formatting changes

TrainingJob readme needs to be updated, docs are upto date now, it doesnt need to have all the setup stuff

File: components/aws/sagemaker/TrainingJob/component.yaml (hunk 1 above)
Comment (surajkota) [reply components/aws/sagemaker/TrainingJob/component.yaml:186]: release version should be 2.1.0, last digit is incremented for bug fixes https://semver.org/

File: components/aws/sagemaker/TrainingJob/src/TrainingJob_spec.py [hunk 2]
```diff
@@ -132,7 +135,7 @@ class SageMakerTrainingJobSpec(
         ),
         output_data_config=InputValidator(
             input_type=SpecInputParsers.yaml_or_json_dict,
-            description="Specifies the path to the S3 location where you want to store model artifacts. Amazon SageMaker crea",
+            description="Specifies the path to the S3 location where you want to store model artifacts. SageMaker creates sub",
```
Comment (surajkota) [inline components/aws/sagemaker/TrainingJob/src/TrainingJob_spec.py:138]: need to fix this later

**Review by surajkota — APPROVED** [review]
/approve

hold until new image is published

**Pull request merged**